cli/cmd/resources/ui.go

```diff
@@ -228,40 +194,40 @@ func NewUIClusterRole() *rbacv1.ClusterRole {
 			Name: "odigos-ui",
 		},
 		Rules: []rbacv1.PolicyRule{
-			{
+			{ // Needed to get and instrument namespaces
 				APIGroups: []string{""},
 				Resources: []string{"namespaces"},
 				Verbs:     []string{"get", "list", "watch", "patch"},
 			},
-			{
+			{ // Needed to get "potential destinations"
 				APIGroups: []string{""},
 				Resources: []string{"services"},
-				Verbs:     []string{"list"},
-			},
-			{
-				APIGroups: []string{""},
-				Resources: []string{"configmaps"},
-				Verbs:     []string{"get", "list", "watch", "patch", "create", "delete", "update"},
+				Verbs:     []string{"get", "list", "watch"},
 			},
-			{
+			{ // Needed for "Describe Source"
 				APIGroups: []string{""},
 				Resources: []string{"pods"},
-				Verbs:     []string{"get", "list"},
+				Verbs:     []string{"get", "list", "watch"},
 			},
-			{
+			{ // Needed for "Describe Source", and for "Describe Odigos"
+				APIGroups: []string{"apps"},
+				Resources: []string{"replicasets"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{ // Needed to instrument applications
 				APIGroups: []string{"apps"},
-				Resources: []string{"deployments", "statefulsets", "daemonsets", "replicasets"},
+				Resources: []string{"deployments", "statefulsets", "daemonsets"},
 				Verbs:     []string{"get", "list", "watch", "patch", "update"},
```

Comment:
one more thing we can clean up. the ui only watches for 3 CRDS [here](https://github.com/odigos-io/odigos/blob/3f17ebd0a20b5bbbc4e37bf17889b26e069f77e8/frontend/main.go#L321).

I think we can safely remove the "watch" permission from all other places. setting up a watcher is an expensive operation, and the permissions makes sure that if we add anything later on - it's intentional.